src/main/java/com/conveyal/datatools/manager/jobs/MergeFeedsJob.java

```diff
@@ -455,33 +455,66 @@ private int constructMergedTable(Table table, List<FeedToMerge> feedsToMerge,
                             fieldsFoundList = Arrays.asList(fieldsFoundInZip);
                         }
                         if (mergeType.equals(MTC) && table.name.equals("stops")) {
-                            // For the first line of the stops table, check that the alt. key
-                            // field (stop_code) is present. If it is not, revert to the original
-                            // key field. This is only pertinent for the MTC merge type.
-                            // TODO: Use more sophisticated check for missing stop_codes than
-                            //  simply the first line containing the value.
-                            if (feedIndex == 0) {
-                                // Check that the first file contains stop_code values.
-                                if ("".equals(keyValue)) {
+                            if (lineNumber == 0) {
+                                // Before reading any lines in stops.txt, first determine whether all records contain
```

Comment:
Suggestion: move to its own method.

Reply:
This is probably fine for a patch, however I see two distinct sections under `if (linenumber == 0)` that could be split.

Reply:
@binh-dam-ibigroup, I don't like the idea of treating this code as different because it's part of a patch. If you have a suggestion for how to elegantly move this into its own method, please create another PR. However, I need to deploy this for now because it is overdue.